openid-4-verifiable-presentations-1_0.md

```diff
@@ -320,6 +320,24 @@ The following is a non-normative example of a transaction data content, after ba
 }
 ```
 
+`attachments`:
+: OPTIONAL. Array of objects, where each object includes an extra resource that needs to be passed to the Wallet. These resources are intended to provide additional context to the Wallet or the End-User. The Wallet MUST ignore any unrecognized attachment formats. Each object contains the following parameters:
+
+    * `format`: REQUIRED: String that identifies the format of the attachment. The specific values are out of scope of this specification. It is RECOMMENDED to use collision-resistant names for type values.
+    * `data`: REQUIRED: The structure and content of data are format-dependent and not constrained by this specification.
+    * `credential_ids`: OPTIONAL. Array of strings each referencing a Credential requested by the Verifier for which the attachment is relevant. Each string matches the `id` field in a DCQL Credential Query. If omitted the attachment is relevant to all requested credentials.
```

Comment:
I think it would be highly beneficial if the wallet would be able to identify an attachment without having to parse it. My assumption right now would be that data and format doesn't uniquely identify an attachment? 

I do believe adding something like an ID that is uniquely defined in an ecosystem on this layer would help to make sure the wallet is not parsing payloads it doesn't need or won't be able to understand?

Reply:
As i understood it, format should be enough, and i think we should encourage that. Attachments are ecosystem / use case specific. 

The `eudi_registration_certificate` already acts like a kind of ID here (And i think we should avoid overloading the id parameter). ID feels more appropriate for an instance of an attachment, not the type of an attachment.

Reply:
I'd agree that ID was probably not the best proposal here.

Let's rename to something like type if we expect this parameter to convey more than a data format imho and make that more clear in the description? I am happy for things to stay the same if this parameter conveys both
- type of the payload (e.g., EUD RP Registration)
- encoding format (e.g. JWT) to allow the Wallet to decode
but it should be a bit more clear in the description imho.

Reply:
why don't we structure this like DCQL queries or VCI requests? The format determines how the type is specified and the rest is determined by the type. I don't think we need an ID as long as the `credential_ids` allow the wallet to determine which attachments shall be used for what context, which is especially important if multiple attachment of same format/type are in the request.

Reply:
type is absolutely fine for me and a better name. When I wrote ID, i basically meant type information - some unique identifier that allows the wallet to figure out what this attachment would be used for.

I agree that if you have type+format information, then the `credential_ids` reference is enough for the required logic 👍

Reply:
type is better than format. But will there be a case where an attachment of a specific type can be represented in different formats? In case of JWT vs. json serialization, it could then be like `eudi_registration_certificate+json` or. `eudi_registration_certificate+jwt`. IMHO I think the amount of the multiple formats are so rar that we should just use the `type` (ex format field` for identification and the wallet will only process the ones it knows.

Reply:
I agree @cre8, i think adding a separate type and format overcomplicates things. The type can mandate the format. And if there need to be multiple formwts we can use +jwt, +cwt, ..